src/main/java/mate/academy/MergeSortAction.java

```diff
@@ -1,8 +1,62 @@
 package mate.academy;
 
-public class MergeSortAction {
+import java.util.Arrays;
+import java.util.concurrent.RecursiveAction;
+
+public class MergeSortAction extends RecursiveAction {
+    private static final int THRESHOLD = 2;
+    private final int[] array;
+    private final int start;
+    private final int end;
 
     public MergeSortAction(int[] array) {
+        this.array = array;
+        this.start = 0;
+        this.end = array.length;
```

Comment:
The `end` variable should be initialized to `array.length - 1` instead of `array.length` to prevent an `ArrayIndexOutOfBoundsException` during the merge operation.